### PR TITLE
GitHub and Office365 improvements to API requests

### PR DESCRIPTION
**Pull request by cborla**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/12984|

## Description

The following changes are introduced:

- The new request collects events using UTC datetime, avoiding local time collection issues.
- Scan time by default changed from 10m to 1m.
- Debug log indicating an API error changed to warning.
- Additionally in the GitHub module, a problem detected with the event type `all` is fixed with a workaround to collect both `git` and `web` events individually.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language

### CODE REVIEW

**Review by vikman90 — APPROVED** [review]

File: src/wazuh_modules/wm_github.c [hunk 1]
```diff
@@ -189,6 +197,8 @@ cJSON *wm_github_dump(const wm_github* github_config) {
 }
 
 STATIC void wm_github_execute_scan(wm_github *github_config, int initial_scan) {
+    int event_types_len = 0;
```
Comment (cborla) [inline src/wazuh_modules/wm_github.c:200]: Could be an unsigned int.

File: src/wazuh_modules/wm_github.c [hunk 2]
```diff
@@ -189,6 +197,8 @@ cJSON *wm_github_dump(const wm_github* github_config) {
...
+    int event_types_it = 0;
```
Comment (cborla) [inline src/wazuh_modules/wm_github.c:201]: Could be an unsigned int.

File: src/wazuh_modules/wm_github.c (hunk 1 above)
Comment (TomasTurina) [reply src/wazuh_modules/wm_github.c]: Done!

File: src/wazuh_modules/wm_github.c (hunk 2 above)
Comment (TomasTurina) [reply src/wazuh_modules/wm_github.c]: Done!

**Review by cborla — COMMENTED** [review]
LGTM!

**Pull request merged**